target/linux/ramips/dts/mt7628an_xiaomi_miwifi-3c.dts

```diff
@@ -0,0 +1,120 @@
+//SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "xiaomi,miwifi-3c", "mediatek,mt7628an-soc";
+	model = "Xiaomi MiWiFi 3C";
+
+	aliases {
+		led-boot = &led_status_amber;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_blue;
+		led-upgrade = &led_status_amber;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_blue: status_blue {
+			label = "blue:status";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_red: status_red {
+			label = "red:status";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_amber: status_amber {
+			label = "amber:status";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "gpio", "refclk", "wdt", "wled_an";
+		function = "gpio";
+	};
+};
+
+&ehci {
+	status = "disabled";
+};
+
+&ohci {
+	status = "disabled";
+};
+
+&esw {
+	mediatek,portmap = <0x2f>;
```

Comment:
This would correspond to 4=wan in switch setup?

Reply:
I looked for documentation the only reference I found was this one. https://patchwork.ozlabs.org/project/lede/patch/20170222143634.GA3820@makrotopia.org/. I change portmap for "llllw". But for portdisable I didn't find anything. 9a8c702

Reply:
For portmap, 0x2f is binary 101111. The zero marks the wan port. So, just need to adjust that so 0 becomes the real wan port, which is 0 based on your configuration (assuming that your configuration is correct).

port-disable seemed correct at first check. It's the same concept, where you will have binary 1 for disable and 0 for enable ...

Reply:
"llllw" is for other platforms (mt7620) ...

Reply:
02_network that is correct and tested.

For portmap, 011111 -> 0x1f
For portdisable, 010101 -> 0x15

It's correct? @adschm

Reply:
Err, no. Would you write "one thousand" as 0001?

Reply:
I write 1000 ahhaha.
Ok. MSB <- LSB.
I need flip bits.

For portmap, 111110 -> 0x3e
For portdisable, 101010 -> 0x2a

It's correct? @adschm

Reply:
Fixed 9a8c702

Reply:
Well, if you resolve it, how should I read the answer?

Reply:
Seems correct now, though.